Parser fails on certain files

<!---
Thanks for filing an issue 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting since someone might have asked the same thing before!

If you are having trouble running stryker, please fill in the template below.
If you have a feature request, feel free to ignore the template.
-->

### Summary

I was trying to implement Stryker for one of our repos to see if it was easy to do. Sadly, I've had no luck so far. I ran into two issues: the first one is that the parser crashes on certain files. But the error stacktrace is not helpful (for an end user), because it starts and ends within the node modules.

This is the Stryker output for such an error:
```
$ stryker run
[2018-02-07 08:46:56.360] [INFO] ConfigReader - Using stryker.conf.js in the current working directory.
[2018-02-07 08:46:57.104] [INFO] InputFileResolver - Found 1 of 161 file(s) to be mutated.
[2018-02-07 08:46:57.105] [INFO] InitialTestExecutor - Starting initial test run. This may take a while.
[2018-02-07 08:46:57.106] [INFO] BabelConfigReader - Reading .babelrc file from path "/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/.babelrc"
[2018-02-07 08:47:13.505] [INFO] InitialTestExecutor - Initial test run succeeded. Ran 118 tests in 16 seconds.
[2018-02-07 08:47:13.509] [ERROR] StrykerCli - an error occurred { SyntaxError: Unexpected token (37:6)
    at Parser.pp$5.raise (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$3.parseIdentifier (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:4332:10)
    at Parser.pp$3.parsePropertyName (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:4156:96)
    at Parser.parsePropertyName (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:6229:23)
    at Parser.pp$3.parseObj (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:4045:12)
    at Parser.pp$3.parseExprAtom (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3719:19)
    at Parser.parseExprAtom (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:7238:22)
    at Parser.pp$3.parseExprSubscripts (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3344:19)
    at Parser.parseMaybeAssign (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:6474:20)
    at Parser.pp$3.parseParenAndDistinguishExpression (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3828:26)
    at Parser.pp$3.parseExprAtom (/Users/ronderksen/Sites/blaze-wishlist-site-wehkamp/node_modules/babylon/lib/index.js:3709:19) pos: 1631, loc: Position { line: 37, column: 6 } }
```

### Stryker config

```js
const path = require('path');

module.exports = function(config) {
  config.set({
    files: [
      {
        pattern: 'src/app/path/to/file.js',
        mutated: true,
        included: false,
      },
      'src/public/**/*',
      'src/app/**/*.scss',
      'src/app/**/path/to/file.spec.js',
      'src/app/**/path/to/file.spec.js.snap',
      'src/**/*',
      'test/*',
      'development/*',
    ],
    testRunner: 'jest',
    mutator: 'javascript',
    transpilers: ['babel'],
    reporter: ['html', 'clear-text', 'progress'],
    coverageAnalysis: 'off',
    babelrcFile: '.babelrc',
    jest: {
      config: require(path.resolve(__dirname, './test/jest.config.json')),
    },
  });
};

```

### Stryker environment

<!-- Please list your stryker plugins + versions here (output of `npm ls | grep stryker`) -->

```
├─┬ stryker@0.18.2
├─┬ stryker-api@0.12.0
├─┬ stryker-babel-transpiler@0.2.4
├─┬ stryker-html-reporter@0.11.5
├─┬ stryker-javascript-mutator@0.3.3
├─┬ stryker-jest-runner@0.4.0
├─┬ stryker-webpack-transpiler@0.1.2
```

### Your Environment
| software         | version(s)
| ---------------- | -------  
| node             |  v8.9.4
| npm              |  v.5.5.1
| Operating System |  macOS 10.13.2

--- a/packages/stryker-javascript-mutator/src/helpers/BabelParser.ts
+++ b/packages/stryker-javascript-mutator/src/helpers/BabelParser.ts
@@ -11,7 +11,11 @@ export default class BabelParser {
       sourceType: 'script',
       plugins: [
         'jsx',
-        'flow'
+        'flow',
+        'objectRestSpread',
+        'classProperties',
+        'asyncGenerators',
+        'dynamicImport'
       ]
     };
 

--- a/packages/stryker-javascript-mutator/test/unit/JavaScriptMutatorSpec.ts
+++ b/packages/stryker-javascript-mutator/test/unit/JavaScriptMutatorSpec.ts
@@ -141,6 +141,41 @@ describe('JavaScriptMutator', () => {
     });
   });
 
+  it('should generate mutants for js vnext code', () => {
+    const sut = new JavaScriptMutator(new Config());
+    const files: File[] = [
+      {
+        name: 'testFile.js',
+        included: false,
+        mutated: true,
+        transpiled: false,
+        kind: FileKind.Text,
+        content: `
+          function objectRestSpread(input) {
+            return {
+              ...input,
+              foo: true,
+            };
+          }
+
+          class ClassProperties { b = 1; }
+
+          async function* asyncGenerators(i) {
+            yield i;
+            yield i + 10;
+          }
+
+          function dynamicImport(){
+            import('./guy').then(a)
+          }
+        `
+      }
+    ];
+
+    const mutants = sut.mutate(files);
+    expect(mutants).lengthOf.above(2);
+  });
+
   it('should generate mutants for multiple files', () => {
     let mutator = new JavaScriptMutator(new Config());
     let file: File = {

